### PR TITLE
fix (sdk): interpolate, replace same var prefix

### LINKED ISSUE #2046: fix(sdk): fix interpolate function
with this testcase

```golang
{
  name: "config with dot after .env",
  args: args{
    input: `
{
"HOST": "customer{{.cds.env.lb.prefix}}.{{.cds.env.lb}}"
}`,
    vars: map[string]string{"cds.env.lb": "lb", "cds.env.lb.prefix": "myprefix"},
  },
  want: `
{
"HOST": "customermyprefix.lb"
}`,
},
```

### PR DESCRIPTION
**Pull request by yesnault**
close #2046

before fix:
Running tool: /Users/yesnault/go/bin/go test -benchmem -run=^$ github.com/ovh/cds/sdk/interpolate -bench ^BenchmarkDo$

goos: darwin
goarch: amd64
pkg: github.com/ovh/cds/sdk/interpolate
BenchmarkDo-4   	   10000	    170029 ns/op	   53574 B/op	     546 allocs/op
PASS
ok  	github.com/ovh/cds/sdk/interpolate	1.732s
Success: Benchmarks passed.

Running tool: /Users/yesnault/go/bin/go test -benchmem -run=^$ github.com/ovh/cds/sdk/interpolate -bench ^BenchmarkDoNothing$

goos: darwin
goarch: amd64
pkg: github.com/ovh/cds/sdk/interpolate
BenchmarkDoNothing-4   	   50000	     30224 ns/op	   11906 B/op	     247 allocs/op
PASS
ok  	github.com/ovh/cds/sdk/interpolate	1.798s
Success: Benchmarks passed.

---
With fix:

goos: darwin
goarch: amd64
pkg: github.com/ovh/cds/sdk/interpolate
BenchmarkDo-4   	   10000	    183572 ns/op	   55689 B/op	     556 allocs/op
PASS
ok  	github.com/ovh/cds/sdk/interpolate	1.867s
Success: Benchmarks passed.

Running tool: /Users/yesnault/go/bin/go test -benchmem -run=^$ github.com/ovh/cds/sdk/interpolate -bench ^BenchmarkDoNothing$

goos: darwin
goarch: amd64
pkg: github.com/ovh/cds/sdk/interpolate
BenchmarkDoNothing-4   	   50000	     30278 ns/op	   11904 B/op	     247 allocs/op
PASS
ok  	github.com/ovh/cds/sdk/interpolate	1.825s
Success: Benchmarks passed.

-----

Signed-off-by: Yvonnick Esnault <yvonnick.esnault@corp.ovh.com>

### CODE REVIEW

File: sdk/interpolate/interpolate.go [hunk 1]
```diff
@@ -20,7 +21,17 @@ func Do(input string, vars map[string]string) (string, error) {
 	defaults := make(map[string]string, len(vars))
 	empty := make(map[string]string, len(vars))
 
-	for k, v := range vars {
+	// sort key, to replace the longer variables before
+	// see "same prefix" unit test
+	var keys sort.StringSlice
+	for k := range vars {
+		keys = append(keys, k)
```
Comment (fsamin) [inline sdk/interpolate/interpolate.go]: do not append, use make

File: sdk/interpolate/interpolate.go [hunk 2]
```diff
@@ -20,7 +21,17 @@ func Do(input string, vars map[string]string) (string, error) {
...
+	}
+	keys.Sort()                      // short to long
+	sort.Sort(sort.Reverse(keys[:])) // reverse, to get the longer before
```
Comment (fsamin) [inline sdk/interpolate/interpolate.go]: should be nice to sort only once

File: sdk/interpolate/interpolate.go [hunk 3]
```diff
@@ -20,7 +27,16 @@ func Do(input string, vars map[string]string) (string, error) {
 	defaults := make(map[string]string, len(vars))
 	empty := make(map[string]string, len(vars))
 
-	for k, v := range vars {
+	// sort key, to replace the longer variables before
+	// see "same prefix" unit test
+	keys := make([]string, len(vars))
+	for k := range vars {
+		keys = append(keys, k)
```
Comment (bnjjj) [inline sdk/interpolate/interpolate.go]: don't use append here because you declare array with make

Comment (yesnault) [reply sdk/interpolate/interpolate.go]: @bnjjj ok, from 549 allocs/op to 548 allocs/op without append :)

**Pull request merged**